dbus/src/message/parser.rs

```diff
@@ -0,0 +1,290 @@
+        let mut match_rule = MatchRule::new();
+
+        for &(key, raw_value) in &self.tokens {
+            let value = self.clean_string(raw_value);
+            match key {
+                "type" => {
+                    match_rule = match_rule.with_type(MessageType::try_from(value.as_str())?);
+                    Ok(())
+                }
+                "interface" => {
+                    match_rule.interface = Some(Interface::new(value).map_err(Error::BadConversion)?);
+                    Ok(())
+                }
+                "sender" => {
+                    match_rule.sender = Some(BusName::new(value).map_err(Error::BadConversion)?);
+                    Ok(())
+                }
+                "member" => {
+                    match_rule.member = Some(Member::new(value).map_err(Error::BadConversion)?);
+                    Ok(())
+                }
+                "path" => {
+                    match_rule.path = Some(Path::new(value).map_err(Error::BadConversion)?);
+                    Ok(())
+                }
+                "path_namespace" => {
+                    match_rule.path = Some(Path::new(value).map_err(Error::BadConversion)?);
+                    match_rule.path_is_namespace = true;
+                    Ok(())
+                }
+                "eavesdrop" => {
+                    match raw_value {
+                        "'true'" | "true" => {
+                            match_rule = match_rule.with_eavesdrop();
+                            Ok(())
+                        }
+                        "'false'" | "false" => {
+                            Ok(())
+                        }
+                        _ => {
+                            Err(Error::BadBoolean)
+                        }
+                    }
+                }
+                _ => {
+                    // Args and Destination are not supported yet.
+                    Err(Error::UnknownKey)
+                }
+            }?;
+        }
+
+        Ok(match_rule)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::message::parser::Error;
+    use crate::message::MatchRule;
+
+    #[test]
+    fn test_tokenizer() -> Result<(), Error> {
+        let mr = MatchRule::parse(r"interface='org.freedesktop.Notifications',member='Notify'")?;
+        assert_eq!(mr.match_str(), "interface='org.freedesktop.Notifications',member='Notify'");
+        let mr = MatchRule::parse(r"interface='org.mpris.MediaPlayer2.Player' , path= /org/mpris/MediaPlayer2,member='Notify', eavesdrop ='true'")?;
+        assert_eq!(mr.match_str(), "path='/org/mpris/MediaPlayer2',interface='org.mpris.MediaPlayer2.Player',member='Notify',eavesdrop='true'");
+        Ok(())
+    }
+
+    #[test]
+    fn test_malformed() {
+        // This panics due to `cstring_wrapper!` not supporting TryInto / TryFrom
+        assert!(MatchRule::parse(r"interface='org.freedesktop.Notifications',member=").is_err());
```

Comment:
I don't understand this comment. Is it still relevant?

Reply:
Oops, forgot to remove that comment.